pandag/nodes.py

```diff
@@ -66,10 +66,11 @@ def update(self, df, loc):
                 df.loc[loc, k] = df.eval(v,
                                          local_dict=self.local_dict,
                                          global_dict=self.global_dict)
-        if self.expr:
+        if self.expr and any(loc):
             # If there was an eval expression specified, update matching rows
             # with it.
             # Expressions can update multiple columns, if separated by newlines
+            # Skips if no rows need to be evaluated.
```

Comment:
do you need a matching loc on the rhs? Because eval_df is made from the whole thing?

eg

```
df.loc[loc, eval_df.columns] = eval_df.loc[loc, :]
```